### PR TITLE
Include caution for property sources

### PR DESCRIPTION
**Pull request by joemccall86**
It is common for java developers to use camel-case in their variable names. Include a caution against using that convention in a configuration, since Micronaut does not translate environment variables for camel-case configuration values.

### CODE REVIEW

File: src/main/docs/guide/config/propertySource.adoc [hunk 1]
```diff
@@ -21,6 +21,8 @@ Micronaut by default contains `PropertySourceLoader` implementations that load p
 
 TIP: `.properties`, `.json`, `.yml` are supported out of the box. For Groovy users `.groovy` is supported as well.
 
+CAUTION: Custom confiruation options that are camel-case (e.g., `fooBar`) may behave unexpectedly when evaluating against the property list above. Specifically, there is no way for an OS environment variable to specify a value for an option named `fooBar`. Micronaut would translate `FOO_BAR` to `foo_bar`, `foo.bar`, but not `fooBar`. Similarly it would translate `FOOBAR` to `foobar` but not `fooBar`.
```
Comment (ilopmar) [inline src/main/docs/guide/config/propertySource.adoc]: Typo in `confiruation`

Comment (joemccall86) [reply src/main/docs/guide/config/propertySource.adoc]: Good catch. Fixed.

**Comment by joemccall86** [comment]
That makes sense. It is indeed well-covered in that section - I don't know how I missed that before. What do you think about this proposed change to outline the consequences of using camel case?

**Comment by jameskleeh** [comment]
@joemccall86 A small blurb saying that environment variables cannot be resolved camel case in that section seems OK to me

**Pull request merged**